xsd/netex_part_1/part1_networkDescription/netex_routeInstruction_version.xsd

```diff
@@ -73,11 +73,22 @@ Rail transport, Roads and Road transport
 		<xsd:complexContent>
 			<xsd:extension base="strictContainmentAggregationStructure">
 				<xsd:sequence>
-					<xsd:element ref="RouteInstruction" maxOccurs="unbounded">
-						<xsd:annotation>
-							<xsd:documentation>An Instruction on how to follow a ROUTE through the network.</xsd:documentation>
-						</xsd:annotation>
-					</xsd:element>
+					<xsd:choice>
```

Comment:
The point here remains, it is now a sequence, choice, sequence. My ideal would be that it would be a choice either being a sequence of RouteInstructionRef or a sequence of RouteInstruction. Now it looks like worst of both worlds. I think it should be either redone (as mentiond before) or point me to other positions where this choice of strictContainmentAggregationStructure is modelled in this way.

Reply:
I don't get your point here Stefan, except the non necessary sequence I pointed above, this is doing what I understand you want to do:


	<netex:routeInstructions>
		<netex:RouteInstructionRef ref="normalizedString">normalizedString</netex:RouteInstructionRef>
		<netex:RouteInstructionRef ref="normalizedString">normalizedString2</netex:RouteInstructionRef>
	</netex:routeInstructions>


works


	<netex:routeInstructions>
		<netex:RouteInstruction>
			<netex:Instruction>toto</netex:Instruction>
			<netex:RoadName>titi</netex:RoadName>
		</netex:RouteInstruction>
		<netex:RouteInstruction>
			<netex:Instruction>hihi</netex:Instruction>
			<netex:RoadName>hoho</netex:RoadName>
		</netex:RouteInstruction>
	</netex:routeInstructions>


Also works
but


	<netex:routeInstructions>
		<netex:RouteInstructionRef ref="normalizedString">normalizedString</netex:RouteInstructionRef>
		<netex:RouteInstructionRef ref="normalizedString">normalizedString2</netex:RouteInstructionRef>
		<netex:RouteInstruction>
			<netex:Instruction>toto</netex:Instruction>
			<netex:RoadName>titi</netex:RoadName>
		</netex:RouteInstruction>
	</netex:routeInstructions>


Does not work (does not validate the XSD)

Reply:
If the initial sequence was retained, wouldn't it allow to:
choice(RouteInstructionRef|RouteInstruction), choice(RouteInstructionRef|RouteInstruction), choice(RouteInstructionRef|RouteInstruction)

instead of
RouteInstructionRef, RouteInstructionRef, RouteInstructionRef 

or

RouteInstruction, RouteInstruction, RouteInstruction,

Reply:
simplfied (but did work before)

Reply:
yes the simplification does not change anything .. just cleaner / more compact !